xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2156,6 +2156,7 @@ void CVideoPlayer::CheckContinuity(CCurrentStream& current, DemuxPacket* pPacket
        fabs(this_dts - that_dts) < DVD_MSEC_TO_TIME(1000))
     {
       m_offset_pts += correction;
+      m_VideoPlayerVideo->SetOffsetPts(m_offset_pts);
```

Comment:
as mentioned this is only the case for overlay already in the container. all demuxer packets including subs are corrected by UpdateTimestamps. UpdateTimestamps is before PTSStartTIme is set.